base/io.jl

```diff
@@ -661,6 +661,31 @@ function readuntil(s::IO, delim::T; keep::Bool=false) where T
     return out
 end
 
+# In readuntil, we use a cache to accelerate scanning of sequences
+# with repeated patterns. Each cache entry tells us which index
+# we should start the search at. When we allocate the cache,
+# we initialize it to 0 (and offset by the first index afterwards).
+# Suppose target is:
+#    Index:  1245689
+#    Value: "aδcaδcx"
+# We would set the cache to
+#    0 0 0 1 0 3 4 0
+# So after if we mismatch after the second aδc sequence,
+# we can pick back up at index 5 (4 + 1).
+function grow_readuntil_cache!(cache, target, max_pos, pos)
```

Comment:
I don't think we should factor out this random chunk of code. It makes it look somehow reusable or independent, but it's neither, so IMO, it just breaks up the code flow.

Reply:
Fair enough, but I didn't find the code very "flowy" in the first place. We could precompute the cache with no change to the rest of the code flow. Would you like it better if it was an inner function?